### PR TITLE
Update CocoaPods to v1.14

### PR DESCRIPTION
**Pull request by huextrat**
## Summary:

CocoaPods release v1.14.0 which fix crash with ActiveSupport >=7.1.x, so it's no longer mandatory to set ActiveSupport version in Gemfile.

Ruby >=2.7 is required by CocoaPods 1.14+

## Changelog:

[iOS][Changed] - Set the CocoaPods version to 1.14.0 as it handle ActiveSupport >=7.1.x

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,6 @@
 source 'https://rubygems.org'
 
 # You may use http://rbenv.org/ or https://rvm.io/ to install and use this version
-ruby ">= 2.6.10"
+ruby ">= 2.7.8"
 
-gem 'cocoapods', '~> 1.12'
-gem 'activesupport', '>= 6.1.7.3', '< 7.1.0'
+gem 'cocoapods', '~> 1.14'
```
Comment (okwasniewski) [inline Gemfile:6]: Looking forward to this change as this version brings support for visionos! 🎉

**Comment by cipolleschi** [comment]
Hi @huextrat, thank you for spending time on the issue.
I discussed the matter internally, and, unfortunately, for the time being we don't want to force all the React Native users to use a newer version of Cocoapods.
* For what concern the Active Support issue, we already released a patch for all the supported React Native versions (0.70 -> 0.73-RC) so the issue is not relevant for our users.
* This change forces to use Ruby 2.7.x and we wanted to keep React Native working with the lowest version of ruby offered by MacOS, which is 2.6.10.

File: Gemfile (hunk 1 above)
Comment (cipolleschi) [reply Gemfile:6]: Actually, VisionOS is supported from [1.13.0](https://github.com/CocoaPods/CocoaPods/releases/tag/1.13.0), I think they backported it somehow.

Comment (okwasniewski) [reply Gemfile:6]: @cipolleschi Do you think we can update it to `1.13.0`? I can open a PR for this 👍🏻

**Pull request closed without merging**